### PR TITLE
cmd/pd-server: emit error detail instead of generic ErrParseFlags

### LINKED ISSUE #3445: pd-server crashes with ErrParseFlags instead of an explanation
### What did you do?
```
$ cat pd.toml
[schedule]
low-space-ratio = 0.01

$ ./bin/pd-server -config=./pd.toml
[2021/02/25 11:26:03.020 -08:00] [FATAL] [main.go:58] ["parse cmd flags error"] [error="[PD:main:ErrParseFlags]parse flags error"] [stack="github.com/pingcap/log.Fatal\n\t/Users/kolbe/Devel/go/pkg/mod/github.com/pingcap/log@v0.0.0-20201112100606-8f1e84a3abc8/global.go:59\nmain.main\n\t/Users/kolbe/Devel/git/pingcap/pd/cmd/pd-server/main.go:58\nruntime.main\n\t/usr/local/Cellar/go/1.15.6/libexec/src/runtime/proc.go:204"]
```

### What did you expect to see?

If the `low-space-ratio` configuration option is invalid, a message should tell me that.

### What version of PD are you using (`pd-server -V`)?
```
Release Version: v4.0.0-rc.2-464-g1f200cbc
Edition: Community
Git Commit Hash: 1f200cbcd64769ff3a5faf3b00bdda4c58171401
Git Branch: master
UTC Build Time:  2021-02-25 06:54:30
```

### PR DESCRIPTION
**Pull request by kolbe**
### What problem does this PR solve?

Fixes #3445 

### What is changed and how it works?

Simply log the error returned by cfg.Parse() instead of substituting errs.ErrParseFlags.

### Check List

Tests

- Manual test (add detailed scripts or steps below)

```
$ cat pd.toml
[schedule]
low-space-ratio = 0.01
```

Before:
```
$ ./bin/pd-server -config=./pd.toml
[2021/02/25 11:26:03.020 -08:00] [FATAL] [main.go:58] ["parse cmd flags error"] [error="[PD:main:ErrParseFlags]parse flags error"] [stack="github.com/pingcap/log.Fatal\n\t/Users/kolbe/Devel/go/pkg/mod/github.com/pingcap/log@v0.0.0-20201112100606-8f1e84a3abc8/global.go:59\nmain.main\n\t/Users/kolbe/Devel/git/pingcap/pd
/cmd/pd-server/main.go:58\nruntime.main\n\t/usr/local/Cellar/go/1.15.6/libexec/src/runtime/proc.go:204"]
```

After:
```
$ ./bin/pd-server -config=./pd.toml
[2021/02/25 11:30:59.343 -08:00] [FATAL] [main.go:58] ["parse cmd flags error"] [error="low-space-ratio should be larger than high-space-ratio"] [errorVerbose="low-space-ratio should be larger than high-space-ratio\ngithub.com/tikv/pd/server/config.(*ScheduleConfig).Validate\n\t/Users/kolbe/Devel/git/pingcap/pd/server
/config/config.go:914\ngithub.com/tikv/pd/server/config.(*ScheduleConfig).adjust\n\t/Users/kolbe/Devel/git/pingcap/pd/server/config/config.go:855\ngithub.com/tikv/pd/server/config.(*Config).Adjust\n\t/Users/kolbe/Devel/git/pingcap/pd/server/config/config.go:557\ngithub.com/tikv/pd/server/config.(*Config).Parse\n\t/Use
rs/kolbe/Devel/git/pingcap/pd/server/config/config.go:408\nmain.main\n\t/Users/kolbe/Devel/git/pingcap/pd/cmd/pd-server/main.go:44\nruntime.main\n\t/usr/local/Cellar/go/1.15.6/libexec/src/runtime/proc.go:204\nruntime.goexit\n\t/usr/local/Cellar/go/1.15.6/libexec/src/runtime/asm_amd64.s:1374"] [stack="github.com/pingca
p/log.Fatal\n\t/Users/kolbe/Devel/go/pkg/mod/github.com/pingcap/log@v0.0.0-20201112100606-8f1e84a3abc8/global.go:59\nmain.main\n\t/Users/kolbe/Devel/git/pingcap/pd/cmd/pd-server/main.go:58\nruntime.main\n\t/usr/local/Cellar/go/1.15.6/libexec/src/runtime/proc.go:204"]
```

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

n/a

Related changes

n/a

### Release note

- No release note

### CODE REVIEW

File: cmd/pd-server/main.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func main() {
 	case flag.ErrHelp:
 		exit(0)
 	default:
-		log.Fatal("parse cmd flags error", errs.ZapError(errs.ErrParseFlags))
```
Comment (JmPotato) [inline cmd/pd-server/main.go:58]: `errs.ErrParseFlags` is no longer needed after being deleted, maybe you should clean it from `pkg/errs/errno.go`, too. The reset LGTM.

**Comment by JmPotato** [comment]
@kolbe Any update?

File: cmd/pd-server/main.go (hunk 1 above)
Comment (kolbe) [reply cmd/pd-server/main.go:58]: Done.

**Comment by kolbe** [comment]
@JmPotato removed errs.ErrParseFlags; PTAL

**Comment by JmPotato** [comment]
> @JmPotato removed errs.ErrParseFlags; PTAL

Please fix the statics check. I think an empty `var ()` won't make the linter happy.

**Review by nolouch — APPROVED** [review]

**Pull request merged**